pkgs/games/MarbleMarcher/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib, stdenv, fetchFromGitHub, runtimeShell, cmake, eigen, sfml, libglvnd }:
+
+let
+  pname = "MarbleMarcher";
+
+  # The game doesn’t have any versions in GitHub.
+  # Just taking a commit from the top of the “master” branch.
+  version = "git-2019-04-22";
+
+  src = fetchFromGitHub {
+    owner = "HackerPoet";
+    repo = pname;
+    rev = "478afb8ea9c894d6d8b961583aa557e2f63d6198";
+    sha256 = "1kpn9wnf14wm7867dp3lhb10065h4vzy101gab18zkphlwl1dbwq";
+  };
+
+  esc = lib.escapeShellArg;
+
+  assets = stdenv.mkDerivation {
+    pname = "${pname}-assets-pack";
+    inherit version src;
+
+    phases = [ "installPhase" ];
+
+    installPhase = ''
+      set -eu || exit
+      mkdir -p -- "$out"/assets
+      cp -- "$src"/assets/* "$out"/assets
+    '';
```

Comment:
```suggestion
    installPhase = ''
      mkdir -p "$out/assets"
      cp "$src/assets/"* "$out/assets"
    '';
```

Reply:
Why?

1. Removing `set -eu` will allow to silently fail for `mkdir`.
2. While we technically can say that `$src` and `$out` have predictable content I take safety-first approach in Bash by default. Interpolating any random values can be interpreted as options by those commands.

Reply:
@SuperSandro2000 I removed `set` flags and instead added `|| exit` patterns at the end of the commands so the errors are not get swiped under the carpet and getting you broken derivation builds. Though since `set -u` went away no typos in the variable names are checked.

Reply:
I kept `--` raw arguments separator as well. Please have a look.

Reply:
nix already does set -e and $out is never empty or everything will fall apart. Also we control $out and the builds are sandboxed so nothing can happen anyway.